src/services/assetService.ts

```diff
@@ -33,9 +33,13 @@ export class AssetService {
         // fileNameParts[0] = "video"
         // fileNameParts[1] = "mp4"
         // fileNameParts[2] = "t=5"
-        const fileNameParts = pathParts[pathParts.length - 1].split(/[\.\?#]/);
-        fileName = fileName || `${fileNameParts[0]}.${fileNameParts[1]}`;
-        const assetFormat = fileNameParts.length >= 2 ? fileNameParts[1] : "";
+        fileName = pathParts[pathParts.length - 1];
```

Comment:
You are overriding the `filename` argument in all cases now... If a filename is specified we should automatically use that one...